Shared/Samples/Sketch on the map/SketchOnTheMapView.swift

```diff
@@ -0,0 +1,289 @@
+    @Published private(set) var canUndo = false
+    
+    /// A Boolean value indicating if the geometry editor can perform a redo.
+    @Published private(set) var canRedo = false
+    
+    /// The currently selected element.
+    @Published private(set) var selection: GeometryEditorElement?
+    
+    /// A Boolean value indicating if the geometry can be saved to a graphics overlay.
+    @Published private(set) var canSave = false
+    
+    /// A Boolean value indicating if the geometry can be cleared from the geometry editor.
+    @Published private(set) var canClearCurrentSketch = false
+    
+    /// A Boolean value indicating if the saved sketches can be cleared.
+    @Published private(set) var canClearSavedSketches = false
+    
+    /// The current geometry of the geometry editor.
+    @Published private(set) var geometry: Geometry? {
+        didSet {
+            canUndo = geometryEditor.canUndo
+            canRedo = geometryEditor.canRedo
+            canClearCurrentSketch = geometry.map { !$0.isEmpty } ?? false
+            canSave = sketchIsValidForSaving
+        }
+    }
+    
+    /// A Boolean value indicating if the geometry editor has started.
+    @Published var isStarted = false
+    
+    /// Creates the geometry menu with a geometry editor.
+    /// - Parameter geometryEditor: The geometry editor that the menu should interact with.
+    /// - Parameter graphicsOverlay: The graphics overlay that is used to save geometries to.
+    init(geometryEditor: GeometryEditor, graphicsOverlay: GraphicsOverlay) {
+        self.geometryEditor = geometryEditor
+        self.graphicsOverlay = graphicsOverlay
+        
+        Task { [weak self, geometryEditor] in
+            for await geometry in geometryEditor.$geometry {
+                self?.geometry = geometry
+            }
+        }
+        Task { [weak self, geometryEditor] in
+            for await selection in geometryEditor.$selectedElement {
+                self?.selection = selection
+            }
+        }
+    }
+    
+    /// Saves the current geometry to the graphics overlay and stops editing.
+    /// Precondition: `canSave`
+    func save() {
+        let geometry = geometryEditor.geometry!
+        let graphic = Graphic(geometry: geometry, symbol: symbol(for: geometry))
+        graphicsOverlay.addGraphic(graphic)
+        stop()
+        canClearSavedSketches = true
+    }
+    
+    /// Clears all the saved sketches on the graphics overlay.
+    func clearSavedSketches() {
+        graphicsOverlay.removeAllGraphics()
+        canClearSavedSketches = false
+    }
+    
+    /// Stops editing with the geometry editor.
+    func stop() {
+        geometryEditor.stop()
+        isStarted = false
+    }
+    
+    /// Returns the symbology for graphics saved to the graphics overlay.
+    /// - Parameter geometry: The geometry of the graphic to be saved.
+    private func symbol(for geometry: Geometry) -> Symbol {
```

Comment:
```suggestion
    /// - Parameter geometry: The geometry of the graphic to be saved.
    /// - Returns: Either a marker or fill symbol depending on the type of provided geometry.
    private func symbol(for geometry: Geometry) -> Symbol {
```
Or any return statement.